packages/js/src/generators/library/library.ts

```diff
@@ -46,22 +46,20 @@ export async function libraryGenerator(
   tree: Tree,
   schema: LibraryGeneratorSchema
 ) {
-  const { layoutDirectory, projectDirectory } = extractLayoutDirectory(
-    schema.directory
-  );
-  schema.directory = projectDirectory;
-  const libsDir = schema.rootProject
-    ? '.'
-    : layoutDirectory ?? getWorkspaceLayout(tree).libsDir;
-  return projectGenerator(tree, schema, libsDir, join(__dirname, './files'));
+  return await libraryGeneratorInternal(tree, {
+    // provide a default nameDirectoryFormat to avoid breaking changes
+    // to external generators invoking this one
+    nameDirectoryFormat: 'derived',
+    ...schema,
+  });
```

Comment:
This fixes the issue for **_this_** generator, but it doesn't fix it if external generators wrap others (e.g. @nx/angular:app). I think we should consider checking the variable name as mentioned earlier.

Reply:
Other generators are not using the helper introduced here yet, so they don't have this issue. The idea is to provide a default value for the exposed generator function consumed by others and have an internal implementation which is the one defined in the generator collection. The rest of the first-party plugins will be updated to do the same In follow-up PRs.